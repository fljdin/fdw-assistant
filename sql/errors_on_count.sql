CREATE TABLE public.dummy1 ();
CREATE TABLE public.dummy2 ();

INSERT INTO config (source, target, condition) VALUES
  ('public.dummy1', 'public.dummy2', 'must fail');

-- copy(1) should fail with a syntax error
SELECT invocation FROM plan() \gexec

-- target public.dummy2 should have a failed state
SELECT target, state, rows, total FROM report WHERE stage_id = 1 ORDER BY job_start;
