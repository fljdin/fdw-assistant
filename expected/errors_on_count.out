CREATE TABLE public.dummy1 ();
CREATE TABLE
CREATE TABLE public.dummy2 ();
CREATE TABLE
INSERT INTO config (source, target, condition) VALUES
  ('public.dummy1', 'public.dummy2', 'must fail');
INSERT 0 1
-- copy(1) should fail with a syntax error
SELECT invocation FROM plan() \gexec
CALL assistant.copy(1);
NOTICE:  Executing: TRUNCATE public.dummy2
NOTICE:  Executing: SELECT count(1) FROM public.dummy1 WHERE must fail
ERROR:  syntax error at or near "fail"
-- target public.dummy2 should have a failed state
SELECT target, state, rows, total FROM report WHERE stage_id = 1 ORDER BY job_start;
    target     | state  | rows | total 
---------------+--------+------+-------
 public.dummy2 | failed |    0 |      
(1 row)

